lib/workers/repository/config-migration/branch/commit-message.ts

```diff
@@ -4,43 +4,51 @@ import type { CommitMessage } from '../../model/commit-message';
 import { CommitMessageFactory } from '../../model/commit-message-factory';
 
 export class ConfigMigrationCommitMessageFactory {
-  private readonly config: RenovateConfig;
+  private commitMessage: string | null = null;
+  private prTitle: string | null = null;
 
-  private readonly configFile: string;
+  constructor(
+    private readonly config: RenovateConfig,
+    private readonly configFile: string
+  ) {}
 
-  constructor(config: RenovateConfig, configFile: string) {
-    this.config = config;
-    this.configFile = configFile;
-  }
-
-  create(): CommitMessage {
+  private create(commitMessageTopic: string): CommitMessage {
     const { commitMessage } = this.config;
 
-    this.config.commitMessageAction =
-      this.config.commitMessageAction === 'Update'
-        ? ''
-        : this.config.commitMessageAction;
-
-    this.config.commitMessageTopic =
-      this.config.commitMessageTopic === 'dependency {{depName}}'
-        ? `Migrate config ${this.configFile}`
-        : this.config.commitMessageTopic;
+    const config = {
+      ...this.config,
+      semanticCommitScope: 'config',
+      commitMessageExtra: '',
+      commitMessageAction: '',
+      commitMessageTopic,
+    };
 
-    this.config.commitMessageExtra = '';
-    this.config.semanticCommitScope = 'config';
-
-    const commitMessageFactory = new CommitMessageFactory(this.config);
+    const commitMessageFactory = new CommitMessageFactory(config);
     const commit = commitMessageFactory.create();
 
     if (commitMessage) {
-      commit.subject = template.compile(commitMessage, {
-        ...this.config,
-        commitMessagePrefix: '',
-      });
+      config.commitMessagePrefix = '';
+      commit.subject = template.compile(commitMessage, config);
     } else {
-      commit.subject = `Migrate config ${this.configFile}`;
+      commit.subject = commitMessageTopic;
     }
 
     return commit;
   }
+
+  getCommitMessage(): string {
+    if (this.commitMessage === null) {
+      this.commitMessage = this.create(
+        `Migrate config ${this.configFile}`
+      ).toString();
+    }
+    return this.commitMessage;
```

Comment:
```suggestion
    return this.create(
        `Migrate config ${this.configFile}`
      ).toString();
```

Do we call `getCommitMessage` multiple times? Does it makes sense to store the message?

Reply:
currently we do not call it more than once, but we do instantiate the class more than once.

Reply:
sorry, forgot to commit suggestion before pushing other related changes :|